x-pack/plugins/ml/public/application/explorer/explorer.tsx

```diff
@@ -418,6 +427,50 @@ export const Explorer: FC<ExplorerUIProps> = ({
     tableData.anomalies?.length > 0;
 
   const hasActiveFilter = isDefined(swimLaneSeverity);
+  const selectedJobIds = Array.isArray(selectedJobs) ? selectedJobs.map((job) => job.id) : [];
+
+  useEffect(() => {
+    if (!noJobsSelected) {
+      const checkIfSourceIndicesHaveGeoField = async () => {
+        const sourceIndicesWithGeoFieldsMap: {
+          [key: string]: { [key: string]: { geoFields: string[]; dataViewId: string } };
+        } = {};
+        // Go through selected jobs
+        if (Array.isArray(selectedJobs)) {
+          await asyncForEach(selectedJobs, async (job) => {
+            // If job has source indices property
+            if (Array.isArray(job.sourceIndices)) {
+              // Check fields for each index to see if it has geo fields
+              await asyncForEach(job.sourceIndices, async (index) => {
+                const dataViewId = await getDataViewIdFromName(index);
+
+                if (dataViewId) {
+                  const dataView = await dataViewsService.get(dataViewId);
```

Comment:
I reckon that active `dataViews` (a map `Record<JobId, DataView>`) should be stored in [AnomalyExplorerCommonState](https://github.com/elastic/kibana/blob/9a056f5286515bdef6b65fab741e27f4d4354d58/x-pack/plugins/ml/public/application/explorer/anomaly_explorer_common_state.ts#L31).

Then your custom hook `useSourceIndicesWithGeoFields ` can utilize it via `useAnomalyExplorerContextValue`.

Reply:
As mentioned before - this can be added to the explorer improvements meta issue 👌